Compile error a an wrong arr type in array filter

<!-- It is advisable to update all relevant modules using `v outdated` and `v install` -->
**V doctor:**
```
OS: linux, Ubuntu 21.10
Processor: 12 cpus, 64bit, little endian, Intel(R) Core(TM) i7-8750H CPU @ 2.20GHz
CC version: cc (Ubuntu 11.2.0-7ubuntu2) 11.2.0

getwd: /var/www/vproject
vmodules: /home/tanguy/.vmodules
vroot: /var/www/vproject/v
vexe: /var/www/vproject/v/v
vexe mtime: 2022-05-03 06:38:58
is vroot writable: true
is vmodules writable: true
V full version: V 0.2.4 4da2908.0e5c1ce

Git version: git version 2.32.0
Git vroot status: weekly.2022.16-130-g0e5c1cee
.git/config present: true
thirdparty/tcc status: thirdparty-linux-amd64 eee03626

```

**What did you do?**
`v -g -o vdbg cmd/v && vdbg /var/www/vproject/test/array_filter_struct_bug.v`
```v
module main

pub struct Row {
pub mut:
	value int
}

fn main() {

	mut list := [Row {value: 1}, Row {value: 2}, Row {value: 3}]

	yo := list.filter('it.value > 2')

	print(yo)
}



```

**What did you expect to see?**

A formatted error explaining arr type is wrong


**What did you see instead?**
```
==================
/tmp/v_1000/array_filter_struct_bug.3329210994601799930.tmp.c:11539: error: invalid operand types for binary operation
/tmp/v_1000/array_filter_struct_bug.3329210994601799930.tmp.c:11539: error: invalid aggregate type for register load
...
==================
(Use `v -cg` to print the entire error message)

builder error: 
==================
C error. This should never happen.

This is a compiler bug, please report it using `v bug file.v`.

https://github.com/vlang/v/issues/new/choose

You can also use #help on Discord: https://discord.gg/vlang

```

--- a/vlib/v/checker/fn.v
+++ b/vlib/v/checker/fn.v
@@ -1778,6 +1778,11 @@ fn (mut c Checker) check_map_and_filter(is_map bool, elem_typ ast.Type, node ast
 				c.error('type mismatch, `$arg_expr.name` must return a bool', arg_expr.pos)
 			}
 		}
+		ast.StringLiteral, ast.StringInterLiteral {
+			if !is_map {
+				c.error('type mismatch, should use e.g. `${node.name}(it > 2)`', arg_expr.pos)
+			}
+		}
 		else {}
 	}
 }

--- a/vlib/v/checker/tests/array_filter_arg_mismatch_err.out
+++ b/vlib/v/checker/tests/array_filter_arg_mismatch_err.out
@@ -1,0 +1,21 @@
+vlib/v/checker/tests/array_filter_arg_mismatch_err.vv:17:20: error: type mismatch, should use e.g. `filter(it > 2)`
+   15 |     }]
+   16 |
+   17 |     a1 := list.filter('it.value > 2')
+      |                       ~~~~~~~~~~~~~~
+   18 |     println(a1)
+   19 |
+vlib/v/checker/tests/array_filter_arg_mismatch_err.vv:20:17: error: type mismatch, should use e.g. `any(it > 2)`
+   18 |     println(a1)
+   19 |
+   20 |     a2 := list.any('it.value > 2')
+      |                    ~~~~~~~~~~~~~~
+   21 |     println(a2)
+   22 |
+vlib/v/checker/tests/array_filter_arg_mismatch_err.vv:23:17: error: type mismatch, should use e.g. `all(it > 2)`
+   21 |     println(a2)
+   22 |
+   23 |     a3 := list.all('it.value > 2')
+      |                    ~~~~~~~~~~~~~~
+   24 |     println(a3)
+   25 | }

--- a/vlib/v/checker/tests/array_filter_arg_mismatch_err.vv
+++ b/vlib/v/checker/tests/array_filter_arg_mismatch_err.vv
@@ -1,0 +1,25 @@
+module main
+
+pub struct Row {
+pub mut:
+	value int
+}
+
+fn main() {
+	mut list := [Row{
+		value: 1
+	}, Row{
+		value: 2
+	}, Row{
+		value: 3
+	}]
+
+	a1 := list.filter('it.value > 2')
+	println(a1)
+
+	a2 := list.any('it.value > 2')
+	println(a2)
+
+	a3 := list.all('it.value > 2')
+	println(a3)
+}

